Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift

```diff
@@ -170,7 +176,9 @@ extension ProjectDescription.Target {
             return nil
         }
 
-        let path = folder.appending(RelativePath(target.path ?? "Sources/\(target.name)"))
+        let path = packageFolder.appending(target.relativePath)
+        let publicHeadersPath = path.appending(target.relativePublicHeadersPath)
+        let moduleMap = try moduleMapGenerator.generate(moduleName: target.name, publicHeadersPath: publicHeadersPath)
```

Comment:
Not sure about this 🤔 
This is quite different from the other mappers we have, which just maps between internal tuist models.

IMO it would be a bit strange to generate a project pointing to a modulemap file which doesn't exists but we know it will be created later.

Also, to separate it we would need to reimplement part of the logic (e.g. to iterate on all targets) to generate the modulemaps. I find it more natural for this piece of logic to be here. The fact that the modulemap is a file and not an Xcode setting is just an "implementation detail" so I don't see it as a reason to separate the 2

Reply:
Thanks for clarifying this @danyf90.

I might have not explained myself correctly, but the suggestion was more around the code organization. The mental model I usually have around mappers is that it's a function that takes an input and returns output (maps). Hence why it was surprising to see side effects like writing modulemaps to disk. 

Let's leave it as that, and if we see we keep adding more side effects to the mapper, we should reconsider a different structure with concerns better separated.